.github/workflows/pythonpublish.yml

```diff
@@ -1,32 +1,58 @@
-# This workflows will upload a Python Package using Twine when a release is created
-# For more information see: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#publishing-to-package-registries
+# Based on 
+# https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflow
 
-name: Upload Python Package
+name: Publish Python Package to PyPI
 
 on:
+  pull_request:     # delete this once workflow tested
+    branches: [main]  # delete this once workflow tested
   release:
     types: [created]
-
+  workflow_dispatch:
+    
 jobs:
-  deploy:
-
+  build:
+    name: Build distribution
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v4
     - name: Set up Python
       uses: actions/setup-python@v5
       with:
-        python-version: 3.9
+        python-version: "3.x"
+  
     - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install build
-        pip install twine
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.pypi_password }}
-      run: |
-        python -m build --outdir dist .
-        twine upload dist/*
\ No newline at end of file
+      run: >-
+        python3 -m 
+        pip install 
+        build
+        --user
+    - name: Build a binary wheel
+      run: python3 -m build
+    - name: Store the distribution package
+      uses: actions/upload-artifact@v4
+      with:
+        name: python-package-distributions
+        path: dist/
+        
+  publish-to-pypi:
+    name: >-
+      Publish to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
```

Comment:
@dr-rodriguez , I don't understand this bit.  If we trigger on release, do we also need this?

Reply:
A release creates a tag, so this is just checking that. We do need it, I think.